### PR TITLE
api: add Conversation API

### PR DESCRIPTION
**Pull request by daixiang0**
# Description

Ref https://github.com/dapr/proposals/pull/59

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
@daixiang0 can you please add a basic integration tests that uses the echo component?

**Comment by daixiang0** [comment]
> @daixiang0 can you please add a basic integration tests that uses the echo component?
> 
> This PR would need to include the echo component registration

Sure.

File: tests/integration/suite/daprd/conversation/http/basic.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	return []framework.Option{
+		framework.WithProcesses(b.daprd),
+	}
+}
+
+func (b *basic) Run(t *testing.T, ctx context.Context) {
+	b.daprd.WaitUntilRunning(t, ctx)
+
+	postURL := fmt.Sprintf("http://localhost:%d/v1.0/conversation/echo", b.daprd.HTTPPort())
```
Comment (yaron2) [inline tests/integration/suite/daprd/conversation/http/basic.go:64]: How is this working if the API is `v1.0-alpha1`? https://github.com/dapr/dapr/pull/7858/files#diff-982843874043a320e810f10e11e97448e5e07c3841388262dcda246d04590ddbR30

Comment (yaron2) [reply tests/integration/suite/daprd/conversation/http/basic.go:64]: Ok, just waiting on the test. Otherwise LGTM

**Review by JoshVanL — APPROVED** [review]

**Comment by yaron2** [comment]
Great work @daixiang0, I'll fix up the integration test

**Pull request merged**